### PR TITLE
refactor: Remove `ImportData` meteor model

### PR DESCRIPTION
**Pull request by KevLehman**
ARCH-918

### CODE REVIEW

File: apps/meteor/app/importer-slack/server/importer.js [hunk 1]
```diff
@@ -235,7 +237,7 @@ export class SlackImporter extends Base {
 			// If we have no slack message yet, then we can insert them instead of upserting
 			this._useUpsert = !Messages.findOne({ _id: /slack\-.*/ });
 
-			zip.forEach((entry) => {
+			const operations = zip.map(async (entry) => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/app/importer-slack/server/importer.js:240]: I'll run some stress tests on this to ensure this change doesn't affect the operation. The zip loop is a very sensitive part of the import process.

Comment (KevLehman) [reply apps/meteor/app/importer-slack/server/importer.js:240]: Go there. I was thinking on changing this to either this async map or a for await of, let me know your results 🤗

File: apps/meteor/server/lib/ldap/Connection.ts [hunk 2]
```diff
@@ -349,9 +349,9 @@ export class LDAPConnection {
 
 				const entries: T[] = [];
 
-				res.on('searchEntry', (entry) => {
+				res.on('searchEntry', async (entry) => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/server/lib/ldap/Connection.ts:352]: ldapjs doesn't support async callbacks. Use `Promise.await` here and let's rework the ldap stuff on a separate PR.

File: apps/meteor/server/lib/ldap/Connection.ts [hunk 3]
```diff
@@ -481,9 +481,9 @@ export class LDAPConnection {
 
 			const entries: T[] = [];
 
-			res.on('searchEntry', (entry) => {
+			res.on('searchEntry', async (entry) => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/server/lib/ldap/Connection.ts:484]: same as above.

File: apps/meteor/server/lib/ldap/Connection.ts [hunk 4]
```diff
@@ -547,9 +547,9 @@ export class LDAPConnection {
 			let entries: T[] = [];
 			const internalPageSize = pageSize * 2;
 
-			res.on('searchEntry', (entry) => {
+			res.on('searchEntry', async (entry) => {
```
Comment (pierre-lehnen-rc) [inline apps/meteor/server/lib/ldap/Connection.ts:550]: same

**Review by pierre-lehnen-rc — CHANGES_REQUESTED** [review]

**Review by pierre-lehnen-rc — COMMENTED** [review]
Actually, can you leave the whole Connection.ts file untouched (adding Promise.awaits elsewhere)?

Instead of reworking the ldap to work asynchronously, I'll split the Virtual Importer from the Base code so that the BaseImporter will be async and Virtual continues to be sync.

**Review by pierre-lehnen-rc — APPROVED** [review]

**Pull request merged**